mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -644,21 +644,29 @@ class LogisticRegression @Since("1.2.0") (
 
     val model = copyValues(new LogisticRegressionModel(uid, coefficientMatrix, interceptVector,
       numClasses, isMultinomial))
-    // TODO: implement summary model for multinomial case
-    val m = if (!isMultinomial) {
-      val (summaryModel, probabilityColName) = model.findSummaryModelAndProbabilityCol()
-      val logRegSummary = new BinaryLogisticRegressionTrainingSummary(
+
+    val (summaryModel, probabilityColName, predictionColName)
+      = model.findSummaryModel()
+    val logRegSummary = if (!isMultinomial) {
```

Comment:
We should change this to `if (numClasses <= 2)` since you can train a multinomial model on 2 classes, and we should still get a binary summary in that case. And of course, please add tests.